front/app/modules/commercial/customizable_homepage_banner/index.tsx

```diff
@@ -3,32 +3,49 @@ import { ModuleConfiguration } from 'utils/moduleUtils';
 import LayoutSetting from './admin/LayoutSetting';
 import TwoColumnLayout from './citizen/TwoColumnLayout';
 import TwoRowLayout from './citizen/TwoRowLayout';
-import FeatureFlag from 'components/FeatureFlag';
 import CTASettings from './admin/CTASettings';
 import CTA from './citizen/CTA';
+import useHomepageSettingsFeatureFlag from 'hooks/useHomepageSettingsFeatureFlag';
+
+const RenderOnFeatureFlag = ({ children }) => {
+  const featureFlag = useHomepageSettingsFeatureFlag({
+    sectionEnabledSettingName: 'customizable_homepage_banner_enabled',
+    appConfigSettingName: 'customizable_homepage_banner',
+  });
+
+  return featureFlag ? <>{children}</> : null;
+};
```

Comment:
Needed a mechanism like this to comply with hook linting rules. This is very similar to FeatureFlag, which I'm deprecating in this PR...

I can't generalize this component because it has the same type error FeatureFlag has (not super important for this PR, but we should find a solution to not duplicate so much code, will log it). Most important, this works if I use it with hard-coded values and recreate in each module (like we're doing with 'customizable_homepage_banner' here).